src/components/filter-fields/DatePicker2.jsx

```diff
@@ -0,0 +1,278 @@
+        />
+        {isShowDropdown ? (
+          <ChevronUp onClick={() => setShowDropdown(false)} className="search__icon cursor-pointer" />
+        ) : (
+          <ChevronDown onClick={() => setShowDropdown(true)} className="search__icon cursor-pointer" />
+        )}
+      </div>
+      <div
+        className={`w-1/2 sm:w-auto mt-1 absolute max-w-3xl select-dropdown ${isShowDropdown ? 'show' : ''}`}
+        onMouseLeave={() => setShowDropdown(false)}
+      >
+        <Form className="datepicker_form" formik={formik}>
+          <div className="sdc">
+            <div className="input-container">
+              <div className="input-day">
+                <input
+                  className="inputDay"
+                  type="text"
+                  size="2"
+                  placeholder="XX"
+                  id="day"
+                  name="day"
+                  onChange={formik.handleChange}
+                  value={formik.values.day}
+                />
+                {!(formik.errors.year || formik.errors.month) ?
+                  formik.errors.day && formik.touched.day && (
+                  <p className="errorMessage">{formik.errors.day}</p>
+                  ) : ''}
+              </div>
+              <div className="dropdown-date">
+                <button type="button" className="dropbtn">
+                  <ChevronDown className="w-4 h-6" />
+                </button>
+                <div className="dropdown-content">
+                  <table>
+                    <tbody>
+                      {createDays().map((row) => (
+                        <tr key={row[0]}>
+                          {row.map((dayNumber) => (
+                            <td key={dayNumber} onClick={() => formik.setFieldValue('day', dayNumber)}>
+                              {dayNumber}
+                            </td>
+                          ))}
+                        </tr>
+                      ))}
+                    </tbody>
+                  </table>
+                </div>
+              </div>
+              <div className="input-month">
+                <input
+                  className="inputMonth"
+                  type="text"
+                  size="15"
+                  placeholder={t('month')}
+                  id="month"
+                  onChange={formik.handleChange}
+                  value={formik.values.month}
+                />
+                {!formik.errors.year ? formik.errors.month && formik.touched.month && (
+                  <p className="errorMessage">{formik.errors.month}</p>
+                ) : ''}
+              </div>
+              <div className="dropdown-date">
+                <button type="button" className="dropbtn">
+                  <ChevronDown className="w-4 h-6" />
+                </button>
+                <div className="dropdown-content">
+                  <table>
+                    <tbody>
+                      {createMonths().map((row) => (
```

Comment:
months can be created one time.
And no need to calculate this, it can be literal